Source/QED/BreitWheelerEngineWrapper.cpp

```diff
@@ -0,0 +1,28 @@
+#include "BreitWheelerEngineWrapper.h"
+//This file provides a wrapper aroud the breit_wheeler engine
+//provided by the PICSAR library
+
+using namespace picsar::multi_physics;
+
+// Functors ==================================
+
+// Initialization of the optical depth
+AMREX_GPU_DEVICE
+amrex::Real
+BreitWheelerGetOpticalDepth::operator() () const
```

Comment:
Shouldn't this be `AMREX_FORCE_INLINE` since this is supposed to be called inside a `ParallelFor` loop?
Also, I'm not sure about this, but shouldn't `WarpXBreitWheelerWrapper::internal_get_optical_depth` be also inlined?

Reply:
You are right.

I've moved the definition of this functor to the header, which is now an ```.hpp``` file.
The definition of the functor now reads as follows:
```
class BreitWheelerGetOpticalDepth
{
public:
    BreitWheelerGetOpticalDepth ()
    {};

    AMREX_GPU_DEVICE 
    AMREX_FORCE_INLINE
    amrex::Real operator() () const
    {
        return WarpXBreitWheelerWrapper::
            internal_get_optical_depth(amrex::Random());
    }
};
``` 

```internal_get_optical_depth``` should be inlined. Indeed, it is declared as follows:
```
         PXRMP_GPU
         PXRMP_FORCE_INLINE
         static _REAL internal_get_optical_depth(_REAL unf_zero_one_minus_epsi);
```

where ```PXRMP_FORCE_INLINE``` is defined as follows:
```
        #if defined(__CUDA_ARCH__)
          #define PXRMP_FORCE_INLINE __forceinline__
        #elif defined(__INTEL_COMPILER)
          #define PXRMP_FORCE_INLINE inline __attribute__((always_inline))
        #elif defined(__clang__)
          #define PXRMP_FORCE_INLINE inline __attribute__((always_inline))
        #elif defined(__GNUC__)
          #define PXRMP_FORCE_INLINE inline __attribute__((always_inline))
        #elif defined(__ibmxl__)
          #define PXRMP_FORCE_INLINE inline __attribute__((always_inline))
        #else
          #define PXRMP_FORCE_INLINE inline
        #endif
```

In the future it would probably be better to provide in PICSAR a way to override the ```PXRMP_FORCE_INLINE``` macro (e.g. ```#define PXRMP_FORCE_INLINE AMREX_FORCE_INLINE```).